### PR TITLE
Remove all usages of MailPoet\Tasks\Sending::getScheduledQueues() [MAILPOET-4367]

### PR DESCRIPTION
**Pull request by lysyjan**
## Description
This pull request starts with https://github.com/mailpoet/mailpoet/commit/cfc359b56fa6b22d5efab158793dc3811b984b57 because I wanted to avoid conflicts.

It should be merged after https://github.com/mailpoet/mailpoet/pull/4305 and rebased on the trunk before.

## QA notes
Make sure scheduled newsletter sending works as expected

## Linked tickets
[MAILPOET-4367]

[MAILPOET-4367]: https://mailpoet.atlassian.net/browse/MAILPOET-4367?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JanJakes** [comment]
@rodrigoprimo Could you pls have a look at this? 🙏 I didn't manage to look at this before my AFK. If that doesn't suit you, feel free to assign it to anybody else. Thanks!

**Comment by rodrigoprimo** [comment]
@JanJakes, sure, I can review this PR. Enjoy your time off.

@lysyjan, would it be ok if I wait until #4305 is merged and this PR rebased to review it? #4305 is already on QA so it is likely to be merged soon.

File: mailpoet/tests/integration/Cron/Workers/SchedulerTest.php
Comment (rodrigoprimo) [inline mailpoet/tests/integration/Cron/Workers/SchedulerTest.php]: Maybe instead of adding the new dependency of the Scheduler class, we can use \MailPoetTest::getServiceWithOverrides() to avoid having to update this test again if a new dependency is added to the same class?

File: mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php
Comment (rodrigoprimo) [inline mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php]: `Sending::getScheduledQueues()` has a default value for `$amount` (`Sending::RESULT_BATCH_SIZE`) and this new method doesn't. I don't it that is necessary, just checking if that was a conscious decision and curious about the reasoning behind it.

**Review by rodrigoprimo — DISMISSED** [review]
Great work, @lysyjan. I left two comments that are not blockers. Feel free to move this PR directly to QA.

File: mailpoet/tests/integration/Cron/Workers/SchedulerTest.php
Comment (lysyjan) [reply mailpoet/tests/integration/Cron/Workers/SchedulerTest.php]: This was implemented in a different PR. Here is the commit: https://github.com/mailpoet/mailpoet/commit/bfc2a7d2afc0e148a3440b80f53d770fa5a10d28

File: mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php
Comment (lysyjan) [reply mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php]: I didn't want to use a constant from a different class as a default value in the repository. In the end, I decided to unify the limit parameter with a similar method `findRunningSendingTasks` from this repository. This change was added as a new commit.

File: mailpoet/tests/integration/Cron/Workers/SchedulerTest.php
Comment (rodrigoprimo) [reply mailpoet/tests/integration/Cron/Workers/SchedulerTest.php]: :+1:

File: mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php
Comment (rodrigoprimo) [reply mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php]: :+1:

**Review by rodrigoprimo — DISMISSED** [review]

**Comment by veljkho** [comment]
Hi @lysyjan 
There are 3 conflicting files found.

**Comment by lysyjan** [comment]
@veljkho Done

**Review by Aschepikov — APPROVED** [review]
This is verified, good job.

**Pull request merged**
